contracts/stake/src/wasm.rs

```diff
@@ -22,66 +23,141 @@ impl StakeContract {
         pk: PublicKey,
         signature: Signature,
         value: u64,
-        created_at: BlockHeight,
         proof: Vec<u8>,
     ) {
-        if value < MINIMUM_STAKE {
-            panic!("The staked value is not within range!");
+        if dusk_abi::caller() != rusk_abi::transfer_contract() {
+            panic!("Can only be called from the transfer contract!");
         }
 
-        let caller = dusk_abi::caller();
-        if caller != rusk_abi::transfer_contract() {
-            panic!("Can only be called from the transfer contract");
+        if value < MINIMUM_STAKE {
+            panic!("The staked value is lower than the minimum amount!");
         }
 
-        let block_height = dusk_abi::block_height();
-        let stake = Stake::new(value, created_at, block_height);
+        // allot a stake to the given key and increment the signature counter
+        let mut stake =
+            self.load_mut_stake(&pk).expect("Failed to query state!");
+
+        let counter = stake.counter();
+
+        stake.increment_counter();
+        stake.insert_amount(value, dusk_abi::block_height());
 
-        let message = Self::stake_sign_message(value, created_at);
-        let signature = rusk_abi::verify_bls_sign(signature, pk, message);
+        // required since we're holding a mutable reference to a stake and
+        // `dusk_abi::transact_raw` requires a mutable reference to the state
+        drop(stake);
```

Comment:
Did you consider to avoid the explicit drop in favour to an implicit inner scope drop? (same for other two occurrences in this file)

Reply:
I considered it. If looked clearer to me this way. If it is the consensus that explicit `drop`s are disallowed - why? - I'll change it.